Refactor

ekohl opened the pull request:
See individual commits. Based on https://github.com/dLobatog/npm2rpm/pull/32 but taken a step further.

ekohl committed:
Refactor file file templating

ehelms committed:
Always use the same spec filename

ehelms committed:
Correct cache dir handling

ekohl committed:
Remove generated trailing whitespace

ehelms commented on default.n2r:
```diff
@@ -32,18 +30,16 @@ $BUILD
+$SYMLINK
 
 %clean
 rm -rf %{buildroot} %{npm_cache_dir}
```
This line should only exist if its bundled and defining npm_cache_dir.

ekohl replied:
Should the buildroot still be removed in ```%clean```? for unbundled? Or is that directory actually inside the buildroot and should the space be removed?

ehelms replied:
Standard practice is to include a buildroot clean command. I think the npm_cache_dir cleanup is relative to whether it's used.

ekohl replied:
I think the default clean is ```rm -rf '%{buildroot}'``` and if we'd place ```%{npm_cache_dir}``` inside the buildroot then it'd automatically clean it up. Do you know if this is correct or am I making things up?

ehelms commented:
Small comment but otherwise looks like a good adaptation of my changes.

ekohl committed:
Remove dupliate nodejs_find_provides_and_requires

This is only needed for unbundled RPMs and is placed by the generator in
$PROVIDES.

dLobatog reviewed (approved):
@ekohl I think the clean command is fine, npm_cache_dir may not be in buildroot, it's always defined as:

```
 %define npm_cache_dir /tmp/npm_cache_%{name}-%{version}-%{release}` shows
```

it's true that on non-bundled packages it may be good to just avoid any references to `npm_cache_dir` as @ehelms said, it's just an useless var. i think I'm going to go with merging, feel free to fix that in another PR

Thanks!
